samples/Java/Basic/SimpleProxy/src/main/java/com.microsoft.azure.eventhubs.samples.SimpleProxy/SimpleProxy.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+package com.microsoft.azure.eventhubs.samples.SimpleProxy;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.microsoft.azure.eventhubs.*;
+
+import java.io.IOException;
+import java.net.*;
+import java.nio.charset.Charset;
+import java.time.Instant;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+
+
+public class SimpleProxy {
+
+    public static void main(String[] args)
+            throws EventHubException, ExecutionException, InterruptedException, IOException {
+
+        String proxyIpAddressStr = "---proxyhostname---";
+        int proxyPort = 3128;
+
+        //set the ProxySelector API; which offers the flexibility to select Proxy Server based on the Target URI.
+        ProxySelector systemDefaultSelector = ProxySelector.getDefault();
+        ProxySelector.setDefault(new ProxySelector() {
+            @Override
+            public List<Proxy> select(URI uri) {
+                if (uri != null
+                        && uri.getHost() != null
+                        && uri.getHost().equalsIgnoreCase("youreventbushost.servicebus.windows.net")) {
+                    LinkedList<Proxy> proxies = new LinkedList<>();
+                    proxies.add(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyIpAddressStr, proxyPort)));
+                    return proxies;
+                }
+
+                // preserve system default selector for the rest
+                return systemDefaultSelector.select(uri);
+            }
+            @Override
+            public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
+                // trace and follow up on why proxy server is down
+                if (uri == null || sa == null || ioe == null) {
+                    throw new IllegalArgumentException("Arguments can't be null.");
+                }
+
+                systemDefaultSelector.connectFailed(uri, sa, ioe);
+
+            }
+        });
+
+        // if the proxy being used, doesn't need any Authentication - "setting Authenticator" step may be omitted
+        Authenticator.setDefault(new Authenticator() {
+            @Override
+            protected PasswordAuthentication getPasswordAuthentication() {
+                if (this.getRequestorType() == RequestorType.PROXY
+                        && this.getRequestingScheme().equalsIgnoreCase("http")
```

Comment:
"http" was a bug. PR 421 https://github.com/Azure/azure-event-hubs-java/pull/421 included in client 2.2.0 changes it to "basic"